redisgraph/query_result.py

```diff
@@ -48,9 +52,18 @@ def __init__(self, graph, response):
         if len(response) == 1:
             self.parse_statistics(response[0])
         else:
-            self.parse_results(response)
+            # start by parsing statistics, see if the reported graph version
+            # matches the one we have
             self.parse_statistics(response[-1])  # Last element.
 
+            if(graph.version != self.graph_version):
+                # graph version miss-match, this is an indication that
+                # the graph schema was modified, sync it.
+                graph.version = self.graph_version
+                graph._refresh_schema()
```

Comment:
I think that refresh schema should return the timestamp retrieved from the last procedure reply. The schema can be changed between two procedure calls but you will update only according to the latest procedure reply.
```suggestion
            graph_version = self.graph_version
            while(graph.version != graph_version):
                # graph version miss-match, this is an indication that
                # the graph schema was modified, sync it.
                graph.version = graph_version
                graph_version = graph._refresh_schema()
```

Reply:
taking your comment to the extreme I think that the mapping used in the original compact reply can become obsolete by the time the first procedure call is made, although the client cache might be in sync with the server the mapping between the current result-set and the cache might be complete incorrect, in which case I suggest reissuing the query